front/app/components/FormBuilder/components/FormBuilderSettings/index.tsx

```diff
@@ -91,19 +91,23 @@ const FormBuilderSettings = ({
   const tabActiveBorder = `4px solid ${colors.primary}`;
   const fieldType = watch(`customFields.${field.index}.input_type`);
 
+  const fieldHasLogic = field.logic.rules && field.logic.rules.length > 0;
```

Comment:
I would move this line right between the comment (line 104) and `isMultiselectWithLogic` (line 105).

Code is easier to read & understand when related code is grouped.